File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main()
{
    string text = get_string("Text: ");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    float L = letters / words * 100;
    float S = sentences / words * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    return 0;
}

int count_letters(string text)
{
    int total = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            total += 1;
        }
    }
    return total;
}

int count_words(string text)
{
    int total = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            total += 1;
        }
    }
    return total;
}

int count_sentences(string text)
{
    int total = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            total += 1;
        }
    }
    return total;
}

File: sentimental-readability/readability.py
from cs50 import get_string
import re

text = get_string("Text: ")
# Count letters
letters = len(re.findall("[a-zA-Z]", text))

# Count sentences
sentences = len(re.findall("[.?!]", text))

# Count words
words = text.count(' ') + 1

L = letters / words * 100
S = sentences / words * 100
num = round(0.0588 * L - 0.296 * S - 15.8)

if num < 1:
    print("Before Grade 1")
elif num >= 16:
    print("Grade 16+")
else:
    print(f"Grade {num}")